apps/shared/WebView.qml

```diff
@@ -319,13 +317,9 @@ WebContainer {
                     break
                 }
                 case "Link:AddSearch": {
-                    // TODO: proper implementation
-                    /*if (searchEngine.searchEngines().indexOf(data.engine.title)) {
-                        searchEngine.url = data.url
-                        searchEngine.href = data.engine.href
-                        searchEngine.title = data.engine.title
-                        searchEngine.addSearchEngine()
-                    }*/
+                    if (SearchEngineModel.searchEngines.indexOf(data.engine.title)) {
```

Comment:
SearchEngineModel.contains(data.engine) instead. No, need to expose that old string list.

Reply:
Actually, addSearchEngine check as well if it's there already and adds if not found.